pkg/proxy/ipvs/proxier.go

```diff
@@ -720,61 +651,88 @@ func (handle *LinuxKernelHandler) GetKernelVersion() (string, error) {
 }
 
 // CanUseIPVSProxier checks if we can use the ipvs Proxier.
-// This is determined by checking if all the required kernel modules can be loaded. It may
-// return an error if it fails to get the kernel modules information without error, in which
-// case it will also return false.
-func CanUseIPVSProxier(handle KernelHandler, ipsetver IPSetVersioner, scheduler string) error {
-	mods, err := handle.GetModules()
-	if err != nil {
-		return fmt.Errorf("error getting installed ipvs required kernel modules: %v", err)
-	}
-	loadModules := sets.NewString()
-	loadModules.Insert(mods...)
-
-	kernelVersionStr, err := handle.GetKernelVersion()
+// The ipset version and the scheduler are checked. If any virtual servers (VS)
+// already exist with the configured scheduler, we just return. Otherwise
```

Comment:
Ohh, fun corner case.  Suppose a user was using IPVS mode with a valid scheduler, and has many VSes.  They then change it to an invalid scheduler and restart kube-proxy.  We detect that they have VSes, but none with this new scheduler.  We try the dummy and it fails.  Should we:

a) Throw an error and abort / panic / exit - existing VSes will work, nothing new will work

b) Fall back on a safe scheduler -  existing VSes will work, new VSes will work but maybe not optimally, users may never see the fact that it failed

c) Fall back on iptables - existing VSes will be abandoned, possibly conflicting with iptables logic, users may never see the fact that it failed

d) (b) or (c) plus events indicating a serious failure, users may STILL never see those unless they go looking

Reply:
For now it is a) if an error is returned from CanUseIPVSProxier(), kube-proxy terminates.

It feels like the right thing to do in CanUseIPVSProxier() is to return an error (as now) and let the caller decide.

It is not simple (I thingk) to fallback to the working scheduler in the CanUseIPVSProxier() function since that mean to alter the global kube-proxy config from a low-level function. So my suggestion is to let it be as-is.

Reply:
I thought kube-proxy would fall-back to iptables, but when I test it terminates. Log messages;
```
E1223 18:48:05.252466     401 server.go:492] "Error running ProxyServer" err="content of the proxy server's configuration file was updated"
E1223 18:48:05.252606     401 run.go:74] "command failed" err="content of the proxy server's configuration file was updated"
(restart...)
I1223 18:48:07.351241     830 proxier.go:685] "Virtual Servers" count=1
I1223 18:48:07.351253     830 proxier.go:687] "Virtual server exists" count=1
I1223 18:48:07.351260     830 proxier.go:695] "No existing VS uses the configured scheduler" scheduler="lc"
E1223 18:48:07.352400     830 proxier.go:716] "Could not create dummy VS" err="no such file or directory" scheduler="lc"
E1223 18:48:07.352419     830 server.go:492] "Error running ProxyServer" err="can't use the IPVS proxier: no such file or directory"
E1223 18:48:07.352432     830 run.go:74] "command failed" err="can't use the IPVS proxier: no such file or directory"
(restart forever...)
```
(the one VS is the kubernetes svc)

Reply:
Yeah, we removed the fallback logic in 1609017f2b5911356656e9edabeac3fb44fca1e5

Reply:
I agree (a) is the best choice